src/main/java/edu/utdallas/davisbase/storage/IndexPage.java

```
@@ -0,0 +1,497 @@
+		} catch (Exception e) {
+		}
+	}
+
+	private static void updateMetaDataRoot(RandomAccessFile file, int pageNo) throws IOException {
+		// TODO Auto-generated method stub
+		file.seek(metaDataRootPageNoOffsetInFile);
+		file.writeInt(pageNo);
+	}
+
+	// splitting leaf page
+	public static int splitLeafPage(RandomAccessFile file, int pageNo) throws IOException {
+		sortKeys(file, pageNo);
+		int newSiblingPageNo = AddLeafPage(file);
+		boolean rootflag = CheckifRootNode(file, pageNo);
+		int parentPageNo;
+		try {
+			if (rootflag) {
+				parentPageNo = AddInteriorPage(file);
+				setPageasRoot(file, parentPageNo);
+				setParent(file, pageNo, parentPageNo);
+			} else {
+				parentPageNo = getParent(file, pageNo);
+			}
+			setParent(file, newSiblingPageNo, parentPageNo);
+			splitLeafNodeData(file, pageNo, parentPageNo, newSiblingPageNo);
+			setParentRightChildPointer(file, parentPageNo, newSiblingPageNo);
+			setRightSiblingPointer(file, newSiblingPageNo, getRightSiblingPointer(file, pageNo));
+			setRightSiblingPointer(file, pageNo, newSiblingPageNo);
+
+		} catch (Exception e) {
+		}
+		return -1;
+	}
+
+	public static void setParent(RandomAccessFile file, int pageNo, int parentPageNo) {
+		long seekParentByte = convertPageNoToFileOffset(pageNo) + PAGE_OFFSET_OF_CELL_PARENT_PAGE_NO;
+		try {
+			file.seek(seekParentByte);
+			file.writeInt(parentPageNo);
+		} catch (Exception e) {
+
+		}
+	}
+
+	public static void splitInteriorPage(RandomAccessFile file, int pageNo) throws IOException {
+		sortKeys(file, pageNo);
+		int newSiblingPageNo = AddInteriorPage(file);
+		boolean rootflag = CheckifRootNode(file, pageNo);
+		int parentPageNo;
+		try {
+			if (rootflag) {
+				parentPageNo = AddInteriorPage(file);
+			} else {
+				parentPageNo = getParent(file, pageNo);
+			}
+			splitInteriorNodeData(file, pageNo, parentPageNo, newSiblingPageNo);
+			setParentRightChildPointer(file, parentPageNo, newSiblingPageNo);
+
+		} catch (Exception e) {
```

Remove this `try`-`catch` block. Do *not* suppress exceptions, much less _all_ exceptions. Instead, let exceptions percolate to the caller.